semantic-analysis/src/Analysis/Analysis/Typecheck.hs

```diff
@@ -54,7 +54,7 @@ data Monotype a
   | Bool
   | Int
   | String
-  | Monotype a :-> Monotype a
+  | [Monotype a] :-> Monotype a
```

Comment:
The change to adopt n-ary functions is present in the domain interface, so all domains have to implement it. Thus, we extend our definition of function monotypes as well.